libqtile/window.py

```diff
@@ -1316,3 +1316,14 @@ def cmd_set_position(self, dx, dy, curx, cury):
                 self.group.layout.focused = index2
                 self.group.layoutAll()
                 break
+
+    def __getstate__(self):
+        d = {}
+        d['group'] = self.group.name
+        d['window'] = self.window.wid
+        return d
+
+    def __setstate__(self, state):
+        self.Wid = state['window']
```

Comment:
Also this doesn't restore the window class back to its original form. After unpickling only a single attribute of the window class remains .The `Wid` which I use later in to get the original window state by using `qtile.windowMap`. Pickling by itself cannot make restore the window state.